### PR TITLE
handle instantaneous rewards outside of reward updates

### LINKED ISSUE #1187: bug with instantaneous rewards in the updates
In the creation of a reward update (the `createRUpd` function), does some account with the instantaneous rewards (`irwds`). The inst. rewards are also saved in the reward update. The inst. rewards can change, however, between when the reward update is calculated and when it is applied, causing the changes to the total reward pot to not be accurate.

We should probably just have the `createRUpd` function do nothing at all with the inst. rewards. The accounting for them should probably just be done entirely inside the `NEWEPOCH` rule.

### PR DESCRIPTION
**Pull request by JaredCorduan**
This PR fixes the instantaneous rewards timing bug in #1187.

The functions `createRUpd` and `applyRUpd` no longer do anything with the instantaneous rewards.  The variable `i_rwd` has been removed from the `RewardUpdate` type.

Applying the instantaneous rewards has been completely quarantined to a new rule, `MIR`. This rule is used in `NEWEPOCH`, just after the reward update is applied, and before `EPOCH` is used.

The "redundant" check that was removed in #1189 has been replaced with a "redundant" check in `NEWEPOCH`. It checks that the reward update is neutral w.r.t. ada.

I will fix the preservation of ada proof in section 15.1 in an upcoming PR.

closes #1187.

### CODE REVIEW

File: shelley/chain-and-ledger/formal-spec/chain.tex [hunk 1]
```diff
@@ -313,7 +421,7 @@ \subsection{New Epoch Transition}
 Figure~\ref{fig:rules:new-epoch} defines the new epoch state transition. It has three rules.
 The first rule describes the change in the case of $e$ being equal to the next epoch
 $e_\ell+ 1$. It also calls the $\mathsf{EPOCH}$ rule and checks that the
-instantaneous rewards stored in the reward update is the same as the instantaneous rewards 
+instantaneous rewards stored in the reward update is the same as the instantaneous rewards
```
Comment (mgudemann) [inline shelley/chain-and-ledger/formal-spec/chain.tex]: is this still true now?

Comment (JaredCorduan) [reply shelley/chain-and-ledger/formal-spec/chain.tex]: nope! there is no `i_rwd`s in the reward update, good catch!

Comment (JaredCorduan) [reply shelley/chain-and-ledger/formal-spec/chain.tex]: I've reworded it.

File: shelley/chain-and-ledger/formal-spec/chain.tex [hunk 2]
```diff
@@ -346,21 +454,27 @@ \subsection{New Epoch Transition}
       e = e_\ell + 1
       &
       \var{ru} \neq \Nothing
-      \\
-      (\wcard,~\wcard,~\wcard,~\wcard,~\var{irwd'})\leteq\var{ru}
       &
-      \var{irwd'} = \getIR{es}
+      (\Delta t,~\Delta r,~\var{rs},~\Delta f)\leteq\var{ru}
+      \\
+      \Delta t+~\Delta r+\left(\sum\limits_{\wcard\mapsto v\in\var{rs}}\right)v+\Delta f=0
```
Comment (javierdiaz72) [inline shelley/chain-and-ledger/formal-spec/chain.tex]: Typo: `v` should be inside the existing big parenthesis.

Comment (JaredCorduan) [reply shelley/chain-and-ledger/formal-spec/chain.tex]: indeed! I get that in whatever change to the spec we make next

**Pull request merged**